lib/octopus/proxy.rb

```diff
@@ -74,6 +74,9 @@ def safe_connection(connection_pool)
         connection_pool.connection.enable_query_cache!
       end
       connection_pool.connection
+    rescue NoMethodError
+      proxy_config.reinitialize_shards
+      retry
```

Comment:
This will infinte retry if something goes wrong, are we sure this won't loop endlessly?

Reply:
Hi @tebayoso, yeah we are sure that this should fix the issue, we already added this to another function https://github.com/orcasnet/octopus/pull/1 this instruction should make the function work, in case it keeps getting an error the this should be revert and perform research on this